gateway/src/test/kotlin/helper/LinearRetryTest.kt

```diff
@@ -0,0 +1,29 @@
+package helper
+
+import dev.kord.gateway.retry.LinearRetry
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.TimeSource
+import kotlin.time.measureTime
+
+class LinearRetryTest {
+
+    @Test
+    fun testLinearity() = runTest {
+        val linearRetry = LinearRetry(1.seconds, 10.seconds, 10)
+        var i = 0
+        val elapsed = TimeSource.Monotonic.measureTime {
```

Comment:
Using `TimeSource.measureTime` won't work, it doesn't use the virtual [`currentTime`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/current-time.html).